WifiManager/impl_lg/WifiManagerState.cpp

```diff
@@ -0,0 +1,235 @@
+         }
+      }
+   }
+   return ret;
+}
+
+uint32_t WifiManagerState::getConnectedSSID(const JsonObject &parameters, JsonObject &response)
+{
+   // this is used by Amazon, but only 'ssid' is used by Amazon app and needs to be returned; the rest is not important
+   LOGINFOMETHOD();
+   std::string ssid;
+   if (fetchSsid(ssid))
+   {
+      // only 'ssid' is used by Amazon app and needs to be returned; the rest can be empty for now
+      response["ssid"] = ssid;
+      response["bssid"] = string("");
+      response["rate"] = string("");
+      response["noise"] = string("");
+      response["security"] = string("");
+      response["signalStrength"] = string("");
+      response["frequency"] = string("");
+      returnResponse(true);
+   }
+   else
+   {
+      returnResponse(false);
+   }
+}
+
+void WifiManagerState::statusChanged(const std::string &interface, InterfaceStatus status)
+{
+   if (interface == getWifiInterfaceName())
+   {
+      updateWifiStatus(status);
+   }
+}
+
+void WifiManagerState::updateWifiStatus(WifiManagerImpl::InterfaceStatus status)
+{
+   bool state_changed = false;
+   auto lookup = statusToState.find(status);
+
+   if (lookup != statusToState.end())
+   {
+      if (lookup->second != m_wifi_state)
+      {
+         m_wifi_state = lookup->second;
+         state_changed = true;
+      }
+   }
+   else
+   {
+      LOGWARN("unknown status: %d", status);
+   }
+
+   if (state_changed)
+   {
+      if (m_wifi_state == WifiState::CONNECTED)
+      {
+         static std::string _;
+         // fetchSsid will raise onSSIDsChanged event in case ssid was changed
+         fetchSsid(_);
+      }
+      // Hardcode 'isLNF' for the moment (at the moment, the same is done in default rdk implementation)
+      WifiManager::getInstance().onWIFIStateChanged(m_wifi_state, false);
+   }
+}
+
+uint32_t WifiManagerState::setEnabled(const JsonObject &parameters, JsonObject &response)
+{
+   LOGINFOMETHOD();
+   returnResponse(false);
+}
+
+uint32_t WifiManagerState::getSupportedSecurityModes(const JsonObject &parameters, JsonObject &response)
+{
+   LOGINFOMETHOD();
+   returnResponse(false);
+}
+
+const std::string WifiManagerState::fetchWifiInterfaceName()
+{
+   DBusClient &dbus = DBusClient::getInstance();
+   std::vector<std::string> interfaces;
+   if (dbus.networkconfig1_GetInterfaces(interfaces))
+   {
+      for (auto &intf : interfaces)
+      {
+         std::string type;
+         if (dbus.networkconfig1_GetParam(intf, "type", type) && type == "wifi")
+         {
+            return intf;
+         }
+      }
+   }
+   else
+   {
+      LOGWARN("failed to fetch interfaces via networkconfig1_GetInterfaces");
+   }
+   return "";
+}
+
+const std::string &WifiManagerState::getWifiInterfaceName()
+{
+   static std::mutex interface_name_mutex;
+   std::lock_guard<std::mutex> lock(interface_name_mutex);
+   static std::string name = WifiManagerState::fetchWifiInterfaceName();
```

Comment:
It is a static value, which is protected only within this function (and it is used outside this function)

Reply:
on another look, no synchro is needed here (local static initialization is thread-safe per standard, and we later return the const &)